src/main/resources/liquibase/crop_changelog/24_5_0.xml

```diff
@@ -104,4 +104,598 @@
+		<comment>
+			Create stored procedure to correct any entryno that are not in sequence
+		</comment>
+		<sql dbms="mysql" splitStatements="false">
+			<![CDATA[
+				CREATE PROCEDURE check_listentry_sequence()
+				BEGIN
+				
+				DROP TABLE IF EXISTS list_seq_table;
+
+				CREATE  TABLE list_seq_table (
+					seq INT(11) NOT NULL AUTO_INCREMENT,
+					entryid INT(11) DEFAULT NULL,
+					listid INT(11) DEFAULT NULL,
+					lrecid INT(11) DEFAULT NULL,
+					PRIMARY KEY (seq),
+					KEY edx (entryid),
+					KEY ldx (lrecid)
+				) ENGINE=INNODB;
+
+
+				INSERT INTO list_seq_table ( entryid, listid, lrecid )
+				SELECT entryid, l.listid, lrecid FROM listdata l , listnms n where l.listid = n.listid ORDER BY listid, entryid ;
+
+				UPDATE listdata l,
+					(SELECT lists.listid, val, ((((max_seq - lists.seq)-(max_seq-min_seq))*(-1))+1)  AS list_seq ,  cnt, lrecid FROM 
+					(SELECT listid , MAX(seq) AS max_seq,MIN(seq) min_seq,MAX(entryid) max_val,MIN(entryid) AS min_val, COUNT(*) AS cnt
+					FROM list_seq_table GROUP BY listid )  AS minmaxlist
+				INNER JOIN 
+				(SELECT listid,seq, entryid AS val, lrecid  FROM list_seq_table) AS lists
+					ON lists.listid = minmaxlist.listid
+					ORDER BY seq
+				) AS ordered_list
+				SET l.entryid = ordered_list.list_seq
+					WHERE l.listid = ordered_list.listid AND l.lrecid = ordered_list.lrecid AND l.entryid<>list_seq;
+					
+				DROP TABLE IF EXISTS list_seq_table;
+				
+				END;
+				]]>
+		</sql>
+    </changeSet>
+	
+	<changeSet  id="v24.5.1-2" author="rowena">
+		<preConditions onFail="MARK_RAN">
+			<!-- Check if dataset check_listentry_table was created -->
+				<sqlCheck expectedResult="1">
+					SELECT
+					COUNT(*)
+					FROM
+					information_schema.routines
+					WHERE
+					routine_schema = DATABASE()
+					AND specific_name = 'check_listentry_sequence';
+				</sqlCheck>
+			
+        </preConditions>
+		<comment>
+			Call stored procedure check_listentry_sequence;
+		</comment>
+		<sql dbms="mysql" splitStatements="true">
+			CALL check_listentry_sequence();
+		</sql>
+	</changeSet>
+
+	<changeSet  id="v24.5.1-3" author="rowena">
+		<preConditions onFail="MARK_RAN">
+			<sqlCheck expectedResult="1">
+				SELECT
+				COUNT(*)
+				FROM
+				information_schema.routines
+				WHERE
+				routine_schema = DATABASE()
+				AND specific_name = 'check_listentry_sequence';
+			</sqlCheck>
+		</preConditions>
+		<comment>Drop procedure check_listentry_sequence</comment>
+		<sql dbms="mysql" splitStatements="false">
+			DROP PROCEDURE check_listentry_sequence;
+		</sql>
+	</changeSet>	
+	
+
+
+	<changeSet id="v24.5.2-1" author="rowena">
+        <preConditions onFail="MARK_RAN">
+			<and>
+			<!-- check if there studies made before 2023 -->
+			<sqlCheck expectedResult="1">
+				SELECT COUNT(*) FROM workbench.workbench_project p
+						INNER JOIN workbench.workbench_crop c
+						WHERE c.crop_name = p.crop_type
+						AND c.db_name = DATABASE()
+						AND YEAR(p.start_date)&lt; 20230000;
```

Comment:
@rvtulod Precondtion should check the stock table for max(uniquename) <> count(uniquename), instead of the project.start_date